docs/reference/bindings/datetime.rst

```diff
@@ -0,0 +1,84 @@
+.. _ref_bindings_datetime:
+
+==================
+Date/Time Handling
+==================
+
+EdgeDB has 6 types related to date and time handling:
+
+* :eql:type:`datetime` (:ref:`binary format <ref_protocol_fmt_datetime>`)
+* :eql:type:`duration` (:ref:`binary format <ref_protocol_fmt_duration>`)
+* :eql:type:`cal::local_datetime`
+  (:ref:`binary format <ref_protocol_fmt_local_datetime>`)
+* :eql:type:`cal::local_date`
+  (:ref:`binary format <ref_protocol_fmt_local_date>`)
+* :eql:type:`cal::relative_duration`
+  (:ref:`binary format <ref_protocol_fmt_relative_duration>`)
+* :eql:type:`cal::date_duration`
+  (:ref:`binary format <ref_protocol_fmt_date_duration>`)
+
+We generally map datetime values to a compatible native type in the binding's
+language when that native type meets all these criteria:
+
+* The type is in the language's standard library
+* The type has sufficient range (EdgeDB timestamps range from year 1 to 9999)
+* The type has sufficient precision (at least microseconds)
+
+If any of the above is not true we tend to provide a custom type in the client
+library itself that is convertible to whatever type language or any popular
+external dependencies have. Exceptions: JavaScript has millisecond precision on
+their ``Date`` type (which is actually a timestamp) and we decided that it's
+better to use that type by default even though it doesn't have enough
+precision.
```

Comment:
Thanks for the clarification, @tailhook. See if this works:

```suggestion
If any of the above criteria are *not* met, we usually provide a custom type
in the client library that can be converted to a language-native type or a 
type provided by a third-party library. Exception: The JavaScript ``Date``
type (which is actually a timestamp) has only millisecond precision, but we
determined it's better to use that type by default even though it doesn't have
sufficient precision.
```

Is there anything we can say here about why we made this call on the JavaScript ``Date`` conversion? I might not go into anything super complicated here, but if we could briefly explain why we made this call, I think it could be useful.

Reply:
Well, more thinking about that, I'm not sure that JavaScript is special. Maybe it's reverse: Rust is special (i.e. Rust has edgedb-specific types, primarily because REPL should represent types exactly). For the datetime type Rust and JavaScript are the only exceptions as far as I remember. I'm not sure about duration/relativedelta.

Reply:
@tailhook Sounds like it might be worth calling out both JavaScript _and_ Rust.

Reply:
I don't think adding more text will help. I think we can say something like "If these criteria don't met, please ping us in chat, or use your best judgement."